springwolf-core/src/main/java/io/github/springwolf/core/asyncapi/scanners/common/AsyncAnnotationScanner.java

```diff
@@ -58,14 +58,27 @@ protected Stream<MethodAndAnnotation<A>> getAnnotatedMethods(Class<?> type) {
         Class<A> annotationClass = this.asyncAnnotationProvider.getAnnotation();
         log.debug("Scanning class \"{}\" for @\"{}\" annotated methods", type.getName(), annotationClass.getName());
 
-        return Arrays.stream(type.getDeclaredMethods())
+        return getInheritedMethods(type).stream()
                 .filter(method -> !method.isBridge())
                 .filter(method -> AnnotationScannerUtil.findAnnotation(annotationClass, method) != null)
                 .peek(method -> log.debug("Mapping method \"{}\" to channels", method.getName()))
                 .flatMap(method -> AnnotationScannerUtil.findAnnotations(annotationClass, method).stream()
                         .map(annotation -> new MethodAndAnnotation<>(method, annotation)));
     }
 
+    /**
+     * Recursively gets the list of declared methods of the class' parent classes
+     * @param type Class to retrieve the list of declared methods
+     * @return List of declared methods
+     */
+    private List<Method> getInheritedMethods(Class<?> type) {
```

Comment:
Could we use org.springframework.util.ReflectionUtils#getAllDeclaredMethods for this?